### ISSUE TITLE
[MRESOLVER-284] BREAKING: Some Sisu parameters needs to be bound

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-resolver#958]
**[Tamas Cservenak](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=cstamas)** opened **[MRESOLVER-284](https://issues.apache.org/jira/browse/MRESOLVER-284?redirect=false)** and commented

This is more a "documentation" or heads-up issue.

Integrators of Resolver using Eclipse Sisu should bind at least following "configuration properties" to `@Parameters`
* `aether.syncContext.named.factory`
* `aether.syncContext.named.nameMapper`

For example how to do it refer here: 

If using plain Eclipse Sisu, see https://www.eclipse.org/sisu/docs/api/org.eclipse.sisu.inject/reference/org/eclipse/sisu/Parameters.html

If using Plexus Shim see here:
https://github.com/apache/maven/pull/831

In short, some components now expect "parameters" to be injected from ctor (so container sourced) instead of session.

Integrators not using Eclipse Sisu are left with hopefully sensible defaults.

---

**Issue Links:**
- [MRESOLVER-291](https://issues.apache.org/jira/browse/MRESOLVER-291) Undo MRESOLVER-284
 (_**"is superceded by"**_)
